extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTask.java

```diff
@@ -412,6 +419,12 @@ public boolean publishSegments(Set<DataSegment> segments, Object commitMetadata)
             throw new ISE("WTF?! Driver attempted to publish invalid metadata[%s].", commitMetadata);
           }
 
+          // Upgrade TaskLocks for all segments
+          for (DataSegment segment: segments) {
+            toolbox.getTaskActionClient().submit(new SetLockCriticalStateAction(segment.getInterval(),
+                                                                                TaskLockCriticalState.UPGRADE));
```

Comment:
I see it uses `SegmentAllocateAction` I can change the check to `contains` instead of `equals`and that should take care of this case hopefully